### PR TITLE
Added default formatters

### PR DESCRIPTION
**Pull request by abzcoding**
# Description

We need to have some default formatters for each lang
Also, there was a bug that wouldn't set the formatter config.

## How Has This Been Tested?

- open a lua file
- run `:Format`
- run `:lua print(vim.inspect(O.formatters.filetype['lua']))`

### CODE REVIEW

File: lua/core/formatter.lua [hunk 1]
```diff
@@ -53,7 +53,7 @@ if not status_ok then
   return
 end
 
-formatter.setup {}
+formatter.setup { formatter.config }
```
Comment (abzcoding) [inline lua/core/formatter.lua:56]: @ChristianChiarulli take a look at this PR and let me know if you like it or not
also without this line, the formatters wouldn't have been set

File: ftplugin/c.lua [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+      exe = "clang-format",
+      --  TODO: append to this for args don't overwrite
+      args = {},
+    }
+  end,
+}
+
+O.formatters.filetype["c"] = clang_formatter
+O.formatters.filetype["cpp"] = clang_formatter
+O.formatters.filetype["cmake"] = clang_formatter
```
Comment (Tastyep) [inline ftplugin/c.lua:13]: I don't think clang-format handles cmake formatting, however, I use [cmake-format](https://github.com/cheshirekow/cmake_format) and it works great

**Review by Tastyep — CHANGES_REQUESTED** [review]
I have tried for lua, golang, python and cpp but only got it working for lua.
Do I need to configure something or should this work out of the box ?

File: ftplugin/c.lua (hunk 2 above)
Comment (abzcoding) [reply ftplugin/c.lua:13]: nice catch 👀 i'm adding correct changes rn

**Comment by abzcoding** [comment]
> I have tried for lua, golang, python, shell, and cpp but only got it working for lua.
> Do I need to configure something or should this work out of the box ?
> 
> When I run :Format I get: no formatter defined for xxx file

@Tastyep can you test it again to see if it works 🥺

**Review by Tastyep — APPROVED** [review]
Great work !
Looking forward to seeing this merged

**Comment by abzcoding** [comment]
the issue with this is that users cannot override the default formatter, i'm working on it atm.

**Comment by Tastyep** [comment]
Could you also please configure a formatter for javascript languages ?

**Comment by abzcoding** [comment]
@Tastyep can you test if it works for `javascript` files as well,
also it first looks for local prettier, if not found then uses global one ( or the one that you define)

**Review by mvllow — APPROVED** [review]

**Comment by ChristianChiarulli** [comment]
before we merge I'd like to test a few things on stream

**Comment by mvllow** [comment]
Possibly a slick suggestion for allowing the user to override prettier args:

```lua
local prettier_args = { "--stdin-filepath", vim.api.nvim_buf_get_name(0) }

for k, v in pairs(O.lang.tsserver.formatter.args) do
	prettier_args[k] = v
end
```

Then in `lv-config.lua` if you have `O.lang.tsserver.formatters.args = { "--single-quote" }` it will be added :)

**Pull request merged**